packages/quantic/force-app/main/default/lwc/quanticHeadlessLoader/__tests__/quanticHeadlessLoader.test.js

```diff
@@ -419,5 +429,85 @@ describe('c/quanticHeadlessLoader', () => {
 
       expect(window.coveoHeadless[testId]).toEqual({});
     });
+  });
+
+  describe('registerToStore', () => {
+    beforeEach(() => {
+      window.coveoHeadless = {
+        [testId]: {
+          bindings: {}
+        }
+      }
+    });
+    
+    it('should return error when store is undefined', () => {
+      initQuanticeStoreTest(testId, undefined);
+      registerToStore(testId, 'someFacets', { label: '', facetId: ''});
+
+      expect(window.coveoHeadless[testId].bindings.store).not.toBeDefined();
+      expect(mockedConsoleError.mock.calls.length).toBe(1);
+    });
+    
+    it('should return error when store is empty', () => {
+      initQuanticeStoreTest(testId, {});
+      registerToStore(testId, 'someFacets', { label: '', facetId: ''});
+
+      expect(window.coveoHeadless[testId].bindings.store).toStrictEqual({});
+      expect(mockedConsoleError.mock.calls.length).toBe(1);
+    });
+
+    it('should register the facet data in the store', () => {
+      initQuanticeStoreTest(testId, { state: { someFacets: {}}});
+      registerToStore(testId, 'someFacets', { label: '', facetId: ''});
+
+      expect(window.coveoHeadless[testId].bindings.store.state.someFacets).toBeDefined();
+      expect(mockedConsoleError.mock.calls.length).toBe(0);
+    });
+
+    it('should not update store with same facetId with new data', () => {
+      initQuanticeStoreTest(testId, { state: { someFacets: {}}});
+
+      registerToStore(testId, 'someFacets', { label: 'some label', facetId: 'facetId'});
+      expect(window.coveoHeadless[testId].bindings.store.state.someFacets.facetId.label).toBe('some label');
+
+      registerToStore(testId, 'someFacets', { label: 'a new label', facetId: 'facetId'});
+      expect(window.coveoHeadless[testId].bindings.store.state.someFacets.facetId.label).toBe('some label');
+    });
+  });
+
+  describe('getFromStore', () => {
+    beforeEach(() => {
+      window.coveoHeadless = {
+        [testId]: {
+          bindings: {}
+        }
+      }
+    });
+
+    it('should return error when store is empty', () => {
```

Comment:
```suggestion
    it('should log an error and return undefined when store is empty', () => {
```
given we're testing that an error is logged, and no error is actually returned.